docs/developer/releasing.md

```diff
@@ -0,0 +1,22 @@
+# Releasing
+
+## Output
+
+### Expected artifacts
+
+1. A container image of the shared cluster-api controller manager
+2. A git tag for providers to use
+
+### Artifact locations
+
+1. The container image is found in the registry `gcr.io/k8s-cluster-api` with an
+   image name of `cluster-api-controller` and a tag that matches the release
+   version. For example, in the `0.0.0-alpha.4` release, the container image
+   location is `gcr.io/k8s-cluster-api/cluster-api-controller:0.0.0-alpha.4`
+
+# Process
+
+1. Push a git tag
```

Comment:
Also need to set the image version in the generated yaml so that when the providers vendor the git tag their will start the correct container image. 

See https://github.com/kubernetes-sigs/cluster-api/pull/635